plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,449 @@
+      bytes: the key, or None if the general.keystore file could not be found or
+          parsed.
+    """
+    general_keystore_entry = self._OpenFileInSameDirectory(
+        file_entry, self._GENERAL_KEYSTORE_FILE, parser_mediator)
+
+    if not general_keystore_entry:
+      return None
+
+    general_keystore_file_object = general_keystore_entry.GetFileObject()
+    general_keystore_data = general_keystore_file_object.read()
+    json_data = json.loads(general_keystore_data)
+
+    if not json_data:
+      parser_mediator.ProduceExtractionWarning(
+          'Unable to parse OneDrive keystore file {0:s}'.format(
+              general_keystore_entry.name))
+      return None
+
+    key = json_data[0].get('Key', None)
+    decoded_key = base64.b64decode(key)
+
+    if len(decoded_key) != 32:
+      return None
+
+    return decoded_key
+
+  def _DeobfuscateStrings(self, extracted_strings, obfuscated_string_map):
+    """Deobfuscate strings extracted from OneDrive logs.
+
+    Args:
+      extracted_strings (list[str]): the list of strings extracted from OneDrive
+          log files.
+      obfuscated_string_map (dict[str, str]): the parsed obfuscated string map.
+
+    Returns:
+      list[str]: the list of deobfuscated strings.
+    """
+    if not obfuscated_string_map:
+      return extracted_strings
+
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = re.split(self._DELIMITERS_REGEX, extracted_string)
+      output_string = []
+      for token in tokens:
+        if token not in self._DELIMITERS:
+          token = obfuscated_string_map.get(token, token)
+        output_string.append(token)
+      output_strings.append(''.join(output_string))
+
+    return output_strings
+
+  def _DecryptStrings(self, extracted_strings, aes_key):
+    """Decrypt strings extracted from OneDrive logs.
+
+    Encrypted strings are serialized as follows:
+    * padded using PKCS7 then encrypted using AES in CBC mode.  The AES key is
+      from the general.keystore file and a null initialization vector.
+    * base64 encoded where the resulting '/' and '+'' characters are replaced
+      with '_' and '-'' respectively.
+
+    Args:
+      extracted_strings (list[str]): the list of strings extracted from OneDrive
+          log files.
+      aes_key (bytes): the AES key.
+
+    Returns:
+      list[str]: the list of decrypted strings.
+    """
+    if not aes_key:
+      return extracted_strings
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = re.split(self._DELIMITERS_REGEX, extracted_string)
+
+      output_string = []
+      for token in tokens:
+        if token in self._DELIMITERS:
+          output_string.append(token)
+          continue
+
+        # minimum block size of base64 encoded AES-CBC is 22 bytes
+        if len(token) < 22:
+          output_string.append(token)
+          continue
+
+        if not all(c in self._BASE64_ALPHABET for c in token):
```

Comment:
don't use abbreviations that have no clear meaning

Reply:
Done